src/identity/components/OrganizationListTab/index.tsx

```diff
@@ -0,0 +1,244 @@
+  }, [])
+
+  // Event Handlers
+  const handleChangePage = (page: number) => {
+    const validatedPage = Math.min(page, totalPages)
+    setCurrentPage(validatedPage)
+    const url = new URL(location.href)
+    url.searchParams.set('page', validatedPage.toString())
+    history.replaceState(null, '', url.toString())
+  }
+
+  const handleFilterOrgs = (searchTerm: string) => {
+    const url = new URL(location.href)
+    url.searchParams.set('searchTerm', searchTerm)
+    history.replaceState(null, '', url.toString())
+    setSearchTerm(searchTerm)
+    handleChangePage(1)
+  }
+
+  const handleSortOrgs = (
+    newSortKey: BucketSortKey,
+    newSortDirection: Sort
+  ) => {
+    const url = new URL(location.href)
+    url.searchParams.set('sortKey', newSortKey)
+    url.searchParams.set('sortDirection', newSortDirection)
+    history.replaceState(null, '', url.toString())
+
+    setSortMethod({
+      sortKey: newSortKey,
+      sortDirection: newSortDirection,
+      sortType: SortTypes.String,
+    })
+    handleChangePage(1)
+  }
+
+  // Load the orgs in the current account, if not yet loaded.
+  useEffect(() => {
+    if (orgsLoadedStatus === RemoteDataState.NotStarted) {
+      dispatch(getQuartzOrganizationsThunk(currentAccountId))
+    }
+  }, [currentAccountId, dispatch, orgsLoadedStatus])
+
+  // Determine whether to display the 'Upgrade' banner.
+  useEffect(() => {
+    const checkOrgCreationAllowance = async () => {
+      try {
+        const {allowed, availableUpgrade} = await fetchOrgCreationAllowance()
+
+        const newOrgAllowance = {
+          availableUpgrade: availableUpgrade,
+          isAtOrgLimit: !allowed,
+          status: RemoteDataState.Done,
+        }
+
+        setOrgAllowance(newOrgAllowance)
+      } catch (err) {
+        setOrgAllowance({...defaultOrgAllowance, status: RemoteDataState.Error})
+
+        dispatch(notify(fetchOrgAllowanceFailed()))
+      }
+    }
+
+    if (orgAllowance.status === RemoteDataState.NotStarted) {
+      checkOrgCreationAllowance()
+    }
+  }, [dispatch, orgAllowance.status])
+
+  // Load search, sort, and pagination settings from URL search params.
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search)
+    const urlSearchTerm = params.get('searchTerm')
+    const urlSortDirection = params.get('sortDirection') as Sort
+    const urlSortKey = params.get('sortKey')
+
+    const validSortDirections = [Sort.Ascending, Sort.Descending, Sort.None]
+
+    if (urlSearchTerm) {
+      setSearchTerm(urlSearchTerm)
+    }
+
+    if (
+      urlSortDirection &&
+      urlSortKey &&
+      validSortDirections.includes(urlSortDirection)
+    ) {
+      setSortMethod({
+        sortKey: urlSortKey,
+        sortDirection: urlSortDirection,
+        sortType: SortTypes.String,
+      })
+    }
+  }, [])
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search)
+    const urlPageNumber = parseInt(params.get('page'), 10)
```

Comment:
I remember you introducing me to this neat function: `Number()`. I think we should be able to use it here as well.

Reply:
Good point. On balance though, I'm inclined to keep `parseInt` here because it's user input (which needs validating) instead of data we're getting from quartz (which is already sanitized and doesn't need validating).

I think `parseInt's` behavior of rounding down decimals, and parsing until the first non-numeric character, helps here, because it will force-return integers in situations where `Number` won't. (If the user messes up and typos a 12.1 or a 12a, they'll still get routed to the page they were _probably_ trying to reach, instead of getting sent back to page 1).

![Screen Shot 2022-10-31 at 5 39 26 PM](https://user-images.githubusercontent.com/91283923/199115948-db434797-6aa4-4fe4-83a6-252c3ed763ff.png)

![Screen Shot 2022-10-31 at 5 39 19 PM](https://user-images.githubusercontent.com/91283923/199116089-1c7a6cea-115e-42e9-b25c-da1c8dfc4640.png)